File: java/removeduplicatesfromsortedlist.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        // Start typing your Java solution below
        // DO NOT write main() function
        ListNode p = head;
        if(p==null) return p;
        if(p.next==null) return p;
        while(p.next!=null){
            if(p.next.val==p.val){
                p.next = p.next.next;
            }else{
                p = p.next;
            }
        }
        return head;
    }
}

File: cpp/RemoveDuplicatesfromSortedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        if(head == NULL) {
            return NULL;
        }
        ListNode* q = head;
        int last = head->val;
        ListNode* p = head->next;
        while(p!=NULL) {
            if(p->val == last) {
                p = p->next;
            } else {
                last = p->val;
                q->next = p;
                q = q->next;
                p = p->next;
            }
        }
        q->next = NULL;
        return head;
    }
};
